docs/docs/development/hardware-integration/physical-layouts.md

```diff
@@ -0,0 +1,429 @@
+
+```dts
+/ {
+    position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete; // Not necessary, but generally recommended if you use position maps
+        layout1: layout1 {
+            physical-layout = <&physical_layout1>;
+            positions = <...>; // List of positions to map
+        };
+        layout2: layout2 {
+            physical-layout = <&physical_layout2>;
+            positions = <...>; // List of positions to map
+        };
+    };
+};
+```
+
+Position maps are optional; if a position map is not present, [ZMK Studio](../../features/studio.md) will automatically determine a (potentially inaccurate) mapping based on the physical key properties. This approach is also used as a fallback for positions not specified in the map if the `complete` property is not set.
+
+See also the [configuration section on position maps](../../config/layout.md#physical-layout-position-map).
+
+:::info
+
+Key positions in the maps are numbered like the keys in your keymap, starting at 0. So the first position in the layout is position `0`, the next key position is `1`, etc.
+
+:::
+
+#### Writing a position map
+
+Start by creating the parent node defining the position map:
+
+```dts
+/ {
+    keypad_lossless_position_map {
+        compatible = "zmk,physical-layout-position-map";
+        complete; // Not necessary, but generally recommended if you use position maps
+
+        // Child node 1 here
+
+        // Child node 2 here
+
+        // ...
+    };
+};
+```
+
+Next, select a layout to be the "reference" layout. It is recommended that this be the layout which has the highest number of keys, as that prevents key mappings from being "lost" when switching back and forth between layouts.
+
+Create the child node for the "reference" layout, and fill the `positions` array by iterating through the keys in your layout in the same order as the matrix transform. For a 2x2 macropad the child node would be
+
+```dts
+macropad_map: macropad {
+    physical-layout = <&macropad_layout>;
+    positions // This is equivalent to `positions = <0 1 2 3>;`, reshaped for readability
+        = < 0  1 >
+        , < 2  3 >;
+};
+```
+
+Next, you'll need to write child nodes for every other layout, deriving the position map from the reference. Start by deleting any keys which aren't present in your layout, then reassign the numbers so that they match the matrix transform. Finally, (if the reference layout has more keys) add numbers back into the "empty"/"spare"/"nonexistent" positions so that both position maps count up to the same number. The below examples show this process more clearly.
```

Comment:
I don't think this makes any sense because we haven't explained what `positions` actually does yet. This is a pretty difficult concept to explain in general, and I can't come up with a way to explain it that I think is very simple, so I think we should try to get as many ideas for this as possible and then see which one people find easiest to understand.

I think of it as the `positions` properties in all the layouts being parallel arrays. The values at the same index in each layout indicate which key positions are the same in each layout. If I switch from layout A to layout B, then the binding at key position `A.positions[i]` in the original layout is transferred to key position `B.positions[i]` in the new layout for every index `i` in the position map. 

Key positions that are past the end of the keymap can also be used in order to store a binding so that it is not lost when switching between layouts with different numbers of keys.

A diagram would probably be very helpful to explaining this too. For example, consider layout A which has three keys and layout B which replaces the first two keys with a 2U key. If the position map is defined like this

```dts
// key positions for layout_a
// +---+---+---+
// | 0 | 1 | 2 |
// +---+---+---+
layout_a {
    ...
    positions = <0 1 2>;
};

// key positions for layout_b
// +-------+---+
// |   0   | 1 |
// +-------+---+
layout_b {
    ...
    positions = <0 2 1>;
};
```
Then this is how the key bindings move when switching between layouts A and B
```
+---+---+---+
+ 0 | 1 | 2 |
+---+---+---+
  ^   ^   ^
  |   |   |
  |   +-- | -------+
  v       v        v
+-------+---+    .....
|   0   | 1 |    . 2 .
+-------+---+    .....
```
Note that the key at position 1 in layout A is moved to an unused position 2 in layout B so that it can be restored when switching back to layout A.